plugins/inventory/gql_inventory.py

```diff
@@ -189,6 +183,44 @@ def add_variable(self, host: str, var: str, var_type: str):
         """
         self.inventory.set_variable(host, var_type, var)
 
+    def create_groups(self, json_data: dict):
+        """Creates groups for host based on `group_by` parameters."""
+        for device in json_data["data"]["devices"]:
+            device_name = device["name"]
+            self.inventory.add_host(device_name)
+            for group_by in self.group_by:
+                group_by_path = group_by.split(".")
+                try:
+                    attr_value = device[group_by_path[0]]
+                except KeyError:
+                    self.display.display(f"Could not find value for {group_by_path[0]} on device {device_name}")
+                    continue
+                for attribute in group_by_path[1:]:
+                    if not isinstance(attr_value, Mapping):
+                        error_path = group_by.split(attribute)[0][:-1]
+                        self.display.display(f"Device {device_name} attribute {error_path} is not a dictionary.")
+                        continue
```

Comment:
```suggestion
                        break
```
I believe this should be a break to continue processing values in group_by, as this should just keep raising an error

Reply:
I updated to implement a while loop so this doesn't apply anymore. Resolving.